Data_Engine/Compute/SplitRequestTreeByType.cs

```diff
@@ -67,7 +67,8 @@ public static List<IRequest> SplitRequestTreeByType(this IRequest request, Type
                 simplified.ExtractTrees(splittingType, extracted, new List<IRequest>());
                 extracted = extracted.Select(x => x.SimplifyRequestTree()).ToList();
 
-                if (simplified.AllRequestsOfType(splittingType).Count == 0)
+                simplified = simplified.SimplifyRequestTree();
+                if (simplified != null)
```

Comment:
This is getting closer but I am confused as to why you need to simplify the tree so many times. Simplifying an already simplified tree should return the exact same result, no ?

![image](https://user-images.githubusercontent.com/16853390/108582054-c80e3000-736b-11eb-9b5e-e29e3806735b.png)

I guess that's kind of fine for now but you should at least have 

```c#
extracted.Add(simplified);
```

and not 

```c#
extracted.Add(simplified.SimplifyRequestTree());
```

since the whole point was to avoid adding a null into extracted, right ?

Reply:
Of course, that call is entirely redundant. I have just pushed a fix.